include/eve/module/real/core/function/regular/simd/x86/convert_128.hpp

```diff
@@ -173,97 +272,42 @@ namespace eve::detail
       }
     }
     //==============================================================================================
-    // Convert to 64 bits integer
+    // Convert to 8 bits integer
     //==============================================================================================
-    else if constexpr( std::is_integral_v<Out> && (sizeof(Out) == 8) )
+    else if constexpr( std::is_integral_v<Out> && (sizeof(Out) == 1) )
     {
+      // NOTE: this looks very similar to the aggregated case: find a way to refactor ?
+
       //============================================================================================
-      // 8 -> 64 bits
-      //============================================================================================
-      if constexpr( std::is_integral_v<In> && (sizeof(In) == 1))
-      {
-        if constexpr( current_api >= sse4_1 )
-        {
-          if constexpr( std::is_signed_v<In> && (N::value <= 2) )
-          {
-            return _mm_cvtepi8_epi64(v0);
-          }
-          else if constexpr( !std::is_signed_v<In>  && (N::value <= 2) )
-          {
-            return _mm_cvtepu8_epi64(v0);
-          }
-          else
-          {
-            return convert_(EVE_RETARGET(simd_), v0, tgt);
-          }
-        }
-        else
-        {
-          return convert_(EVE_RETARGET(simd_), v0, tgt);
-        }
-      }
-      //============================================================================================
-      // 16 -> 64 bits
+      // 16 -> 8 bits
       //============================================================================================
-      if constexpr( std::is_integral_v<In> && (sizeof(In) == 2))
+      if constexpr( sizeof(In) == 2 && (N::value <= 8) )
       {
-        if constexpr( current_api >= sse4_1 )
+              if constexpr(N::value == 1) return wide<Out,N>(v0.storage());
+        else  if constexpr(N::value <= 4)
         {
-          if constexpr( std::is_signed_v<In> && (N::value <= 2) )
-          {
-            return _mm_cvtepi16_epi64(v0);
-          }
-          else if constexpr( !std::is_signed_v<In>  && (N::value <= 2) )
-          {
-            return _mm_cvtepu16_epi64(v0);
-          }
-          else
-          {
-            return convert_(EVE_RETARGET(simd_), v0, tgt);
-          }
+          v0 &= static_cast<In>(0x00FFU);
```

Comment:
Why the mask? Did you test and you need it?
I mean we have pack signed and pack unsigned.